Enable saving of settings in the context of current user

ramanbuttar opened the pull request:
If settings are to be saved in the context of `current_user`, specify an additional option `current_user_context` to be `true`.

`def current_user` can be overridden in your own `ApplicationController` to return a different variable than `@current_user`.

Also, collect `settings_class` as a class name constant, instead of a `string`.

accessd commented:
Please exclude 6d0c8bc from here. Also we need tests and entry to CHANGELOG for it. Thank you!

ramanbuttar commented:
* Removed https://github.com/accessd/rails-settings-ui/commit/6d0c8bc96509b3d92be43a90b40277a8675cbea2 from this PR
* Updated version
* Updated change log


* Having some issues with setting up tests so some help would be appreciated
